src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/special/SpecialWikiActivityPageObject.java

```diff
@@ -65,117 +74,78 @@ public void verifyNewWallThreadEntry(String postTitle, String postContent, Strin
-                            + userName + "', not found", false, driver);
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifDetailsPresent(editActivitiesList.get(i), articleName, userName);
+      break;
     }
+
+    return ifPassed;
   }
 
-  /**
-   * verifies if there is the wanted NewPage on WikiActivity, searching through 5 recent editions.
-   */
-  public void verifyRecentNewPage(String articleName, String userName) {
+  public Boolean doesLastNRecentActivitiesContain(int n, String articleName, String userName) {
     Boolean ifPassed = false;
-    for (int i = 0; i < 4; i++) {
-      ifPassed = ifDetailsPresent(newPageActivitiesList.get(i), articleName,
-                                  userName);
-      if (ifPassed) {
-        PageObjectLogging.log("verifyRecentNewPage",
-                              "WikiActivity module found recent new page: '" + articleName
-                              + "' that was created by user: " + userName, true);
-        break;
-      }
-    }
-    if (!ifPassed) {
-      PageObjectLogging.log("verifyRecentNewPage",
-                            "new page:  '" + articleName + "' by user: '"
-                            + userName + "', not found", false, driver);
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifDetailsPresent(activitiesList.get(i), articleName, userName);
+      break;
     }
 
+    return ifPassed;
   }
 
-  /**
-   * verifies if there is the wanted new Blog on WikiActivity, searching through 5 recent editions.
-   */
-  public void verifyRecentNewBlogPage(String blogContent, String blogTitle, String userName) {
+  public Boolean doesLastNRecentBlogActivitiesContain(int n, String blogPostContent, String blogPostName,
+                                                      String userName) {
     Boolean ifPassed = false;
-    for (int i = 0; i < 4; i++) {
-      ifPassed = ifNewBlogDetailsPresent(newPageActivitiesList.get(i), blogContent, blogTitle,
-                                         userName);
-      if (ifPassed) {
-        PageObjectLogging.log("verifyRecentNewBlogPage",
-                              "WikiActivity module found recent new blog titled as: '" + blogTitle
-                              + "' that was created by user: " + userName
-                              + " and contains content: " + blogContent, true);
-        break;
-      }
-    }
-    if (!ifPassed) {
-      PageObjectLogging.log("verifyRecentNewBlogPage",
-                            "WikiActivity module didn't find recent new blog titled as: '"
-                            + blogTitle
-                            + "' that was created by user: " + userName
-                            + " and containted content: " + blogContent, false, driver);
+    for (int i = 0; i < n; i++) {
+      ifPassed = ifNewBlogDetailsPresent(activitiesList.get(i), blogPostContent, blogPostName, userName);
+      if (ifPassed == true) break;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00121)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00122)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1125)